docs/cookbook/using-docker.md

```diff
@@ -0,0 +1,343 @@
+
+::: note
+This means that you **must** commit the project `package-lock.json` and you must not list anything required to build the project assets as a "dev" dependency.  Your project doesn't need any of those in production, right? 
+:::
+
+Following the dependency install, all of the necessary project files are copied into the container. Note that we will also create a `.dockerignore` file to exclude some files and folders from being copied.
+
+Next, we run a script to trigger the apostrophe asset build (we will cover this script next). To set a unique `APOS_RELEASE_ID` environment variable each time we change files and redeploy, we are using the script to create a `release-id` file with a unique string. Otherwise, we would have to change this string each time manually.
+
+Everything until this point helped build the container image. Those commands only run once when the container image is built or rebuilt. The final `CMD` line is what runs every time the container is started.
+
+### Creating the install script
+
+The alpine linux distribution is slim and doesn't include bash, but we can access a shell at `/bin/sh`. Into the `scripts` folder at the root of your project create the following file:
+
+<AposCodeBlock>
+
+```bash
+#!/bin/sh
+
+export APOS_RELEASE_ID=`cat /dev/urandom |env LC_CTYPE=C tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1`
+
+echo $APOS_RELEASE_ID > ./release-id
+
+node app @apostrophecms/asset:build
+```
+
+<template v-slot:caption>
+  scripts/build-asset.sh
+</template>
+
+</AposCodeBlock>
+
+We won't go through this file in detail. As covered in the previous section, it creates a random unique string and copies it out to the `release-id` file at the root of the project. It then triggers the `@apostrophecms/asset` module to build the assets. That module will read the `release-id` file and use the string in the build.
+
+Building the assets inside this script, which is part of a **build step** in the Dockerfile, ensures the assets become part of the image, so they don't have to be re-generated every time the image is used. The same is true for the `release-id` file, which Apostrophe uses to identify the asset bundle it should be using. It'll be the same bundle at build time and at run time. If the image is rebuilt, we'll get a new image, new CSS URLs, and no stale stylesheets.
+
+### Creating a `.dockerignore` file
+The `.dockerignore` file prevents specific files from being copied into your final image. This is important to block sensitive or unnecessary files from being incorporated into your image. Simply go through your directory and copy any file or folder name not needed to build your project into your `.dockerignore` file. Note that folder names are followed by a `/`. I'm using Visual Studio Code in this tutorial, so the topmost folder listed won't be in your project if you use a different editor.
+
+<AposCodeBlock>
+
+```bash
+.vscode/
+apos-build/
+badges/
+data/
+node-modules/
+public/apos-frontend/
```

Comment:
Do not ignore public/apos-frontend. This is the asset build.

Reply:
We build it as part of the container build, but we get no benefit from that if we don't include it in the container.

Reply:
Deleted.